kernel/mutex.c

```diff
@@ -203,12 +203,24 @@ static inline int z_vrfy_k_mutex_lock(struct k_mutex *mutex, s32_t timeout)
 #include <syscalls/k_mutex_lock_mrsh.c>
 #endif
 
-void z_impl_k_mutex_unlock(struct k_mutex *mutex)
+int z_impl_k_mutex_unlock(struct k_mutex *mutex)
 {
 	struct k_thread *new_owner;
 
-	__ASSERT(mutex->lock_count > 0U, "");
-	__ASSERT(mutex->owner == _current, "");
+	/*
```

Comment:
I think permission check should be first so it won't leak information about a non-owned mutex.